src/main/java/me/xmrvizzy/skyblocker/skyblock/FairySouls.java

```diff
@@ -163,10 +170,14 @@ private static void markClosestFairyFound() {
             LOGGER.warn("Failed to mark closest fairy soul as found because player is null.");
             return;
         }
-        fairySouls.get(Utils.getLocationRaw()).stream().filter(FairySouls::isFairySoulNotFound).min(Comparator.comparingDouble(fairySoul -> fairySoul.getSquaredDistance(player.getPos()))).ifPresent(fairySoul -> {
-            initializeFoundFairiesForCurrentProfileAndLocation();
-            foundFairies.get(Utils.getProfile()).get(Utils.getLocationRaw()).add(fairySoul);
-        });
+        fairySouls.get(Utils.getLocationRaw()).stream()
+                .filter(FairySouls::isFairySoulNotFound)
+                .min(Comparator.comparingDouble(fairySoulPos -> fairySoulPos.getSquaredDistance(player.getPos())))
+                .filter(fairySoulPos -> fairySoulPos.getSquaredDistance(player.getPos()) <= 10)
```

Comment:
```suggestion
                .filter(fairySoulPos -> fairySoulPos.getSquaredDistance(player.getPos()) <= 36)
```
The maximum reach is 4.5 blocks and with a bit of margin of error 6 blocks is safe. I don't think there are two fairy souls in close range.

Reply:
Normally a higher value like 36 would make sense but in Skyblock clicking a fairy soul has a lower leach than 4.5.
The highest getSquaredDistance I measured was 9.5.
But you are right that this should be increased a bit, how about 18?

Reply:
Let's do 16.

Reply:
👍